### PR TITLE
proper popups handling

### PR DESCRIPTION
**Pull request by pg83**
Fix for https://github.com/qtile/qtile/issues/4565

### CODE REVIEW

**Comment by jwijenbergh** [comment]
Thanks for the PR. Maybe we also need to check if there is no more parent or of the role is None? See e.g. https://github.com/djpohly/dwl/blob/d6fabe3a150d2e464078c3269410d56e8bbeb54f/client.h#L91-L104.

File: libqtile/backend/wayland/core.py [hunk 1]
```diff
@@ -923,6 +923,8 @@ def _on_xdg_activation_v1_request_activate(
         surface = event.surface
         if surface and surface.is_xdg_surface:
             xdg_surface = XdgSurface.from_surface(surface)
```
Comment (jwijenbergh) [inline libqtile/backend/wayland/core.py:925]: Maybe we also have to check if this is None? Other compositors seem to do it

Comment (jwijenbergh) [reply libqtile/backend/wayland/core.py:925]: I done that in #4571, you can rebase if that is merged

**Comment by pg83** [comment]
> Thanks for the PR. Maybe we also need to check if there is no more parent or of the role is None? See e.g. https://github.com/djpohly/dwl/blob/d6fabe3a150d2e464078c3269410d56e8bbeb54f/client.h#L91-L104.

I would prefer to keep the original version, because this is what I have tested, and for which I can be responsible.

If I have any more errors on this topic in my log, I will come and fix them again!

**Pull request closed without merging**